src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -546,61 +546,113 @@ internal async void ExecutePackageDownload(string name, PackageVersion package,
                 var immediateUninstalls = new List<Package>();
                 var builtinPackages = new List<Package>();
 
-                // if a package is already installed we need to uninstall it, allowing
-                // the user to cancel if they do not want to uninstall the package
-                var duplicateLocalPackages = dependencyVersionHeaders.Select(dep => localPkgs.FirstOrDefault(v => v.Name == dep.name));
-                foreach (var localPkg in duplicateLocalPackages)
-                {
-                    if (localPkg == null) continue;
+                // if the new package has one or more dependencies that are already installed
+                // we need to either first uninstall them, or allow the user to forcibly install the package,
+                // or cancel the installation if they do not want either of the first two options.
+
+                // local package that conflicts with package itself.
+                Package duplicatePackage = null;
+
+                // list of local packages that conflict (have different versions) with package dependencies.
+                var localPkgsConflictingWithPkgDeps = new List<Package>();
+                var newPackageHeaders = new List<PackageVersion>();
 
-                    if (localPkg.BuiltInPackage)
+                foreach(var header in dependencyVersionHeaders)
+                {
+                    bool isHeaderDuplicate = false;
+                    foreach(var pkg in localPkgs)
                     {
-                        builtinPackages.Add(localPkg);
-                        continue;
+                        if (pkg.Name == header.name)
```

Comment:
just a style thing but it might be more understandable if these were called something like:
pkg -> localpkg
and header -> dependencyHeader